lib/HALInterfaces/src/IMqttClient.h

```diff
@@ -139,18 +139,20 @@ class IMqttClient
     /**
      * Subscribes to a topic.
      *
-     * @param[in] topic     Topic to subscribe to. The Client ID is used as base topic: `Client ID`/`topic`
+     * @param[in] topic     Topic to subscribe to.
+     * @param[in] useClientBaseTopic   If true, the client ID is used as the base (prefix) of the topic.
      * @param[in] callback  Callback function, which is called on a new message.
      * @return If successfully subscribed, returns true. Otherwise, false.
      */
-    virtual bool subscribe(const String& topic, TopicCallback callback) = 0;
+    virtual bool subscribe(const String& topic, const bool useClientBaseTopic, TopicCallback callback) = 0;

```

Comment:
Now its no generic MQTT client anymore. Shouldn't that be done on higher level?

Reply:
I believe it is the other way around. Before, the client ID was used always, which makes it not-generic MQTT Client. It could ONLY subscribe to topics of the form `clientID/topic`

Now it is optional to include the client ID as base topic, so it can subscribe to any topic.

It is left as an option to use the client ID for publishing and subscribing, as the application may not know the MQTT client ID